src/projections.jl

```diff
@@ -407,28 +418,126 @@ function projection_on_set(::DefaultDistance, v::AbstractVector{T}, s::MOI.Power
+function _in_pow_cone(v::AbstractVector{T}, cone::MOI.DualPowerCone; tol=DEFAULT_POWER_CONE_TOL_IN_CONE) where {T}
     α = cone.exponent
     return (
-        v[1] >= 0 && v[2] >=0 && tol + (v[1])^α * (v[2])^(1-α) >= α^α * (1-α)^(1-α) * abs(v[3])
+        v[1] >= 0 && v[2] >= 0 && tol + (v[1])^α * (v[2])^(1-α) >= α^α * (1-α)^(1-α) * abs(v[3])
     )
 end
 
+function _power_cone_system_factor(xi, αi, z, r)
+    # If `xi` is negative and `r` is very close to `abs(z)`,
+    # we get to a point where `√(x² + a) + x` is zero.
+    # Even if the difference is small, we would like to
+    # get a more accurate value than zero.
+    # In order to achieve this, we use the following trick:
+    # √(x² + a) + x = (x^2 + a - x^2) / √(x^2 + a) - x
+    xi2 = xi^2
+    a = 4*αi*r*(abs(z) - r)
+    root = xi2 + a
+    if xi > 0
+        return xi + sqrt(root)
+    else
+        return a / (sqrt(root) - xi)
+    end
+end
+
+function _power_cone_system(r, px, py, α)
+    return 0.5 * (px^α * py^(1-α)) - r
+end
+
+function _power_cone_system(r, x, y, z, α)
+    return _power_cone_system(
+        r,
+        _power_cone_system_factor(x, α, z, r),
+        _power_cone_system_factor(y, 1 - α, z, r),
+        α,
+    )
+end
+
+function _solve_system_power_cone_bisection(
+    v::AbstractVector{T},
+    s::MOI.PowerCone;
+    max_iters=DEFAULT_POWER_CONE_MAX_ITERS_BISSECTION,
+    tol=DEFAULT_POWER_CONE_TOL_CONV
+) where {T}
+    x, y, z = v
+    α = s.exponent
+    # Φ is positive for r = 0 and negative for r = |z| so we can just bisect
+    pos = zero(T)
+    neg = abs(z)
+    r = (pos + neg) / 2
+    for _ in 1:max_iters
+        Φ = _power_cone_system(r, x, y, z, α)
+        if Φ < 0
+            neg = r
+        else
+            pos = r
+        end
+        r = (pos + neg) / 2
+        if abs(Φ) < tol
+            break
+        end
+    end
+    println(pos)
+    println(neg)
+    return r
+end
+
+function _solve_system_power_cone_newton(
+    v::AbstractVector{T},
+    s::MOI.PowerCone;
+    max_iters=DEFAULT_POWER_CONE_MAX_ITERS_NEWTON,
+    tol=DEFAULT_POWER_CONE_TOL_CONV
+) where {T}
+    x, y, z = v
+    α = s.exponent
+    # Solve with Newton method
+    # Start Newton at |z|/2. Sol in set (0, |z|)
+    dΦ_prod_dr(xi,αi,z,r,px) = 2*αi*(abs(z) - 2r) / (px - xi)
+    dΦ_dr(r,phi,px,py,dpx,dpy) = (phi+r) * (α * dpx / px + (1-α) * dpy / py) - 1
+    px, py = zero(T), zero(T)
+    r = abs(z) * 0.46
+    for _ in 1:max_iters
+        if isnan(r)
+            break
+        end
+        px = _power_cone_system_factor(x, α, z, r)
+        py = _power_cone_system_factor(y, 1 - α, z, r)
+        Φ = _power_cone_system(r, px, py, α)
+        if abs(Φ) < tol
+            break
+        end
+        dpx = dΦ_prod_dr(x, α, z, r, px)
+        dpy = dΦ_prod_dr(y, 1 - α, z, r, py)
+        dΦ = dΦ_dr(r, Φ, px, py, dpx, dpy)
+
+        # Newton step, bounded to interval
+        r = min(max(r - Φ/dΦ, 0), abs(z))
```

Comment:
sign was off here

Reply:
Ah yes, I played around but forgot to put it back